src/components/MoneyRequestConfirmationList.js

```diff
@@ -518,26 +556,45 @@ function MoneyRequestConfirmationList(props) {
                     isAuthTokenRequired={!_.isEmpty(receiptThumbnail)}
                 />
             )}
-            {shouldShowSmartScanFields && (
+            {props.shouldShowSmartScanFields && (
                 <MenuItemWithTopDescription
                     shouldShowRightIcon={!props.isReadOnly && !props.isDistanceRequest}
                     title={formattedAmount}
                     description={translate('iou.amount')}
-                    onPress={() => !props.isDistanceRequest && Navigation.navigate(ROUTES.MONEY_REQUEST_AMOUNT.getRoute(props.iouType, props.reportID))}
+                    interactive={!props.isReadOnly}
+                    onPress={() => {
+                        if (props.isDistanceRequest) {
+                            return;
+                        }
```

Comment:
@youssef-lr I have a minor question: why do we avoid user edit amount in case of distance request? Please help answer once you have a chance.